fix(client): improve headers layout

lorenzo-cavazzi opened the pull request:
This PR addresses a few layout details in the new headers.

* Restore the missing title in the default avatar on the projects and datasets detail pages.
* Add minimal margin to entity tags to prevent collapsing into the description (in the image, the previous version with the collapsed content)
    ![Screenshot_20230103_144933](https://user-images.githubusercontent.com/43481553/210372653-8699ffa9-e388-4120-b28d-d9a6bfbdcb2d.png)
* Remove extra space on the detail pages when tags are missing. This happened only on small and medium window sizes (in the image, the previous version with the extra space on the bottom)
    ![Screenshot_20230103_145107](https://user-images.githubusercontent.com/43481553/210372887-2a593d85-bc91-4291-9247-08a70ba89338.png)


/deploy renku=renku-ui-2.15.0-tests #persist #cypress

andre-code reviewed (dismissed):
It works as expected and the code looks good🚀.

lorenzo-cavazzi commented:
Tests failed because the default avatar's text overflowed the parent container (see image).
I fixed it and added the same fix to the entities lists -- it wasn't creating issues there cause it was overflowing only vertically, but it's better to keep it clean.

**Before**
![Screenshot_20230104_085452](https://user-images.githubusercontent.com/43481553/210510501-5aedf6df-0828-4b60-a92d-87c26129064b.png)

**After**
![Screenshot_20230104_085806](https://user-images.githubusercontent.com/43481553/210510538-2a46ff1e-7e02-4d3c-9b61-8ea5a1221e72.png)

ciyer reviewed (approved):
Code and appearance look good!
